core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -951,6 +965,36 @@ private[spark] class BlockManager(
     None
   }
 
+  /**
+   * Reads the block from the local directories of another executor which runs on the same host.
+   */
+  private def readDiskBlockFromSameHostExecutor(
+      blockId: BlockId,
+      localDirs: Array[String],
+      blockSize: Long): Option[ManagedBuffer] = {
+    val file = ExecutorDiskReader.getFile(localDirs, subDirsPerLocalDir, blockId.name)
+    if (file.exists()) {
```

Comment:
Imran already commented on this, but it would be good to know what happens if this file disappears after this point. Because `EncryptedManagedBuffer`/`EncryptedBlockData` will not open the file right away, so you may get some exception later, and it would be good to know that the rest of the code will recover properly from that.

(I guess it wouldn't be worse than getting an `IOException` while reading your own cached block, but doesn't hurt to check.)